sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExpressions.scala

```diff
@@ -118,7 +118,7 @@ abstract class GpuUnaryExpression extends UnaryExpression with GpuExpression {
   def outputTypeOverride: DType = null
 
   override def columnarEval(batch: ColumnarBatch): Any = {
-    val input = child.asInstanceOf[GpuExpression].columnarEval(batch)
+    val input = child.columnarEval(batch)
```

Comment:
Can you please clarify. From what I understand so far children could possibly contain a `SortOrder`??

Reply:
Nothing that is evaluated can contain a `SortOrder`.  `SortOrder` is `Unevaluable`. So it will not work columnar or otherwise.  It is up to spark to only insert it into places that it knows will work properly, like `SortExec`.